pkg/store/models/counterparty.go

```diff
@@ -134,6 +139,37 @@ func (c *Counterparty) HasContact(email string) (bool, error) {
 	return false, nil
 }
 
+// Returns the Website as a string after it has been parsed with `url.Parse`,
+// attempting to detect missing schemas and other parsing errors. This function
+// will return an error if Website.String cannot be parsed by url.Parse or if
+// Website.Valid is false.
+func (c *Counterparty) NormalizedWebsite() (out string, err error) {
+	if c.Website.Valid {
+		var parsed *url.URL
+		if parsed, err = url.Parse(c.Website.String); err != nil {
+			return "", err
+		}
+
+		// This is a HACK but it's the best we can do right now. When there is no
+		// schema, then usually the hostname is put into the schema and there is no
+		// Host value, so we check for this and if so we will add the schema and
+		// try again
+		if parsed.Host == "" {
+			if parsed, err = url.Parse("https://" + c.Website.String); err != nil {
+				return "", err
+			}
+		}
+
+		return parsed.String(), nil
+	} else {
+		return "", errors.ErrInvalidString
+	}
```

Comment:
Be nice to your compiler ;-)

```suggestion
	}
	
	return "", errors.ErrNullString
```